src/main/perf/SearchPerfTest.java

```diff
@@ -595,4 +610,13 @@ public void run() {
     out.close();
   }
 
+  private static IndexSearcher createIndexSearcher(IndexReader reader, boolean doConcurrentReads, ExecutorService executorService) {
```

Comment:
Can we remove `doConcurrentReads` here?  And just use `executorService != null` to mean concurrent searching.